### ISSUE TITLE
Setting of mysql dialect only through plugin configuration

### ISSUE THREAD

**Issue by StefanSchubert** [avianey/hsqldb-maven-plugin#1]
Hi there,

first of all thanks for this plugin!  I want to push the flyway db scripts onto CI without requiring a real mysql. You plugin would does the trick. Hower I have not seen a configuration property in maven to set the mysql dialect. Have I overseen something here.

My example setup:

```
    <profile>
        <id>db_test</id>
        <activation>
            <activeByDefault>false</activeByDefault>
        </activation>
        <build>
            <plugins>

                <plugin>
                    <!-- current version -->
                    <groupId>fr.avianey.mojo</groupId>
                    <artifactId>hsqldb-maven-plugin</artifactId>
                    <version>1.0.0</version>

                    <!--
                        default value for in memory jdbc:hsqldb:hsql://localhost/xdb
                        override only values you want to change
                    -->
                    <configuration>
                        <driver>org.hsqldb.jdbcDriver</driver>
                        <path>mem:test</path>
                        <address>localhost</address>
                        <name>xdb</name>
                        <username>sa</username>
                        <password></password>
                        <validationQuery>SELECT 1 FROM INFORMATION_SCHEMA.SYSTEM_USERS</validationQuery>
                        <detail>true</detail>
                    </configuration>

                    <!-- call start and stop -->
                    <executions>
                        <execution>
                            <id>start-hsqldb</id>
                            <phase>pre-integration-test</phase>
                            <goals>
                                <goal>start</goal>
                            </goals>
                        </execution>
                        <execution>
                            <id>stop-hsqldb</id>
                            <phase>post-integration-test</phase>
                            <goals>
                                <goal>stop</goal>
                            </goals>
                        </execution>
                    </executions>

                </plugin>
                <plugin>
                    <groupId>org.flywaydb</groupId>
                    <artifactId>flyway-maven-plugin</artifactId>
                    <version>3.1</version>
                    <configuration>
                        <url>jdbc:hsqldb:hsql://localhost/xdb</url>
                        <user>SA</user>
                        <password/>
                    </configuration>
                    <executions>
                        <execution>
                            <id>clean and update database</id>
                            <phase>integration-test</phase>
                            <goals>
                                <goal>clean</goal>
                                <goal>init</goal>
                                <goal>migrate</goal>
                            </goals>
                        </execution>
                    </executions>
                    <dependencies>
                        <dependency>
                            <groupId>org.hsqldb</groupId>
                            <artifactId>hsqldb</artifactId>
                            <version>2.2.9</version>
                        </dependency>
                    </dependencies>
                </plugin>

            </plugins>
        </build>
    </profile>
```

Thanks for help!

**Comment by StefanSchubert** [comment]
I did some investigations and found out I mixed some things up. Being used to set the database dialect via hibernate or eclipselink I thought is is some property which is supported be the underlying DB.
Instead the mapping logic is performed by the ORM.  Which means in my case I need either flyway translation the scripts to hsql (which will not happen) or find a mysql in-memory db if there is any.

Close the ticked and thanks for listening ;-)
